### DISCUSSION TITLE
loading pretrained resent model in Unet encoder

### DISCUSSION THREAD

**Discussion by ramchandracheke** [qubvel/segmentation_models.pytorch#427] [General]
Hi,
Thanks for this amazing repo. I am wondering, how can I load pretrained model(resnet) in Unet configuration. Currently only imagenet weights are supported.
Thanks

**Comment by qubvel** [comment]
Try model.encoder.load_state_dict()

**Reply by ramchandracheke** [reply]
Thanks King for quick reply!

**Reply by ramchandracheke** [reply]
Thanks @qubvel . That works. There were few extra state_dict in the old_resnet and name was also different. I posted question and solution at : https://discuss.pytorch.org/t/how-to-load-load-state-dict-for-resnet-model/125998
